core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -230,6 +233,8 @@ private void verifyUserDoesNotExist(String mobileNumber)
 
 	private void verifyUserIsAllowed(String mobileNumber)
 	{
+		whitelistedNumberService.validateNumber(mobileNumber);
```

Comment:
See earlier comment on property. If you move this call down under ``if (maxUsers <= 0)``, it's taken care of.